### PR TITLE
easier daemon and client configuration for custom chains

### PR DESCRIPTION
**Pull request by LeeSmet**
This is only a WIP to get some feedback.

Daemon code: 
- Moved all of the code from the rivined package into a rivined/daemon subpackage so it is importable.
- Moved the code which sets up cobra from main to an exported function in this subpackage
- The default variables for the daemon can be set by a config struct which must be passed to the function
- A DefaultConfig method is provided which returns a config as it is hard coded right now

Chain constants:
- These are already exported in the types package so I didn't touch those. Only the calculation of some of them has been moved from the init function in the types package to an exported function (also called at the end of init). 

Considerations: 
- As a result of #93, the daemon package is probably not feasible anymore, so it will need to move to another place, perhaps as another package in the project root? 
- Currently the commands are not exported, which would allow some more freedom when creating a daemon (like setting up your own flags), and then bind to these exported methods to prevent code duplication. Do we want to support this, as there is (currently) not a lot which could be changed imo?

### CODE REVIEW

**Comment by LeeSmet** [comment]
Changes have been applied to the client and daemon, in the client i exported the commands so a custom client can be created which does not support all default commands should this be required

File: rivinec/sort_test.go [hunk 1]
```diff
@@ -0,0 +1 @@
+package rivinec
```
Comment (GlenDC) [inline rivinec/sort_test.go]: can be deleted?

File: cmd/rivinec/daemoncmd.go [hunk 2]
```diff
@@ -1,79 +0,0 @@
...
-	Available bool   `json:"available"`
-	Version   string `json:"version"`
-}
-
-// stopcmd is the handler for the command `siac stop`.
-// Stops the daemon.
-func stopcmd() {
-	err := get("/daemon/stop")
-	if err != nil {
-		die("Could not stop daemon:", err)
```
[the remark this answers is not in the document]
Comment (GlenDC) [reply cmd/rivinec/daemoncmd.go]: Have put this one on the backlog (https://github.com/rivine/rivine/issues/121), as it's not really important enough, for now, to spend more than 5 minutes on this.

File: rivinec/sort_test.go (hunk 1 above)
Comment (GlenDC) [reply rivinec/sort_test.go]: Resolved in a24cac7.

File: rivinec/defaultclient.go [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+	var apiErr api.Error
+	err := json.NewDecoder(resp.Body).Decode(&apiErr)
+	if err != nil {
+		return err
+	}
+	return apiErr
+}
+
+// ApiGet wraps a GET request with a status code check, such that if the GET does
+// not return 2xx, the error will be read and returned. The response body is
```
[the remark this answers is not in the document]
Comment (GlenDC) [reply rivinec/defaultclient.go]: Resolved in a24cac7 (as well as the `ApiPost` call).

File: cmd/rivinec/daemoncmd.go [hunk 4]
```diff
@@ -1,79 +0,0 @@
...
-
-type updateInfo struct {
-	Available bool   `json:"available"`
-	Version   string `json:"version"`
-}
-
-// stopcmd is the handler for the command `siac stop`.
-// Stops the daemon.
-func stopcmd() {
-	err := get("/daemon/stop")
```
[the remark this answers is not in the document]
Comment (GlenDC) [reply cmd/rivinec/daemoncmd.go]: Resolved in a24cac7.

**Pull request merged**